### ISSUE TITLE
semantic: does not flag illegal `super` in various positions in object literals

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#13323]
All of the below are illegal positions for `super`. Semantic does not flag an error for any of them.

```js
let _obj = {
  foo: function() {
    super.foo;
  },
};

let _obj2 = {
  foo() {
    return function() {
      super.foo;
    };
  },
};

let _obj3 = {
  [ function() { super.foo; } ]() {},
  get [ function() { super.foo; } ]() {},
  set [ function() { super.foo; } ](v) {},
};
```

[Playground](https://playground.oxc.rs/#eNp1j09PwzAMxb9K5BNIFeKPuBRxQ7uC4MgQcjN3CkrtykkG09TvTrKSMRCcart5v/feDiy04CmaV+nezK3ZLdmYXqQ1fWIbnfDJ6Xw0JqSR9Cz/vCn71Cx5ytOSq/zySP+tUopJ+Q/cb2BG/gO+quDnH5gjvZnMSzlVLTQg0O5AE5dP2HLED2ijJmrAO451DlZGOizboRNft6jIoRcdoO3RB5oaGFEDaSGi9/L+uK92n2JwK1p8JavyUSm/3dADKnGYrxlRzPeIPB8cZmZEXVNOBhQuzy+ucwcXxGOk1R1Zj4oFn0k1jZUVrWlfkBg7T0+S1NKA48FtcOx6V/2scFTxixy9iDaknYTcfgaWQNkcQ4TpE0+ct8A=)

Related to #13284.

**Comment by overlookmotel** [comment]
PR to fix coming shortly.
